vochain/account.go

```diff
@@ -42,15 +44,15 @@ func (a *Account) Transfer(dest *Account, amount uint64, nonce uint32) error {
 	if a.Nonce != nonce {
 		return ErrAccountNonceInvalid
 	}
-	if a.Balance < amount {
+	a.Nonce++
+	if (a.Balance + cost) < amount {
 		return ErrNotEnoughBalance
 	}
 	if dest.Balance+amount <= dest.Balance {
 		return ErrBalanceOverflow
 	}
 	dest.Balance += amount
-	a.Balance -= amount
-	a.Nonce++
+	a.Balance -= amount + cost
```

Comment:
The spec design said the cost of a transaction should be sent to 0xFFF... burn address.

That will be useful to have within the state always the number of tokens that has been used. Then we can use this number to mint more tokens and keep always the same liquid.

So I'd not add the cost here, I'd just let this method to transfer among amounts. The caller function (tx related) should be the one checking the balance+cost and making to transfers, one to 0xFFF and the other to the destination account.

This way in some moment we can also just switch 0xFFF by the validator address if we want to reward it.